cmd/clusterctl/pkg/client/upgrade.go

```diff
@@ -0,0 +1,47 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+func (c *clusterctlClient) UpgradePlan(options UpgradePlanOptions) ([]UpgradePlan, error) {
+	// Get the client for interacting with the management cluster.
+	cluster, err := c.clusterClientFactory(options.Kubeconfig)
+	if err != nil {
+		return nil, err
+	}
+
+	// Ensures the custom resource definitions required by clusterctl are in place.
+	if err := cluster.ProviderInventory().EnsureCustomResourceDefinitions(); err != nil {
+		return nil, err
+	}
+
+	upgradePlan, err := cluster.ProviderUpgrader().Plan()
+	if err != nil {
+		return nil, err
+	}
+
+	// UpgradePlan is an alias for cluster.UpgradePlan; this makes the conversion
```

Comment:
icky but ok i guess

Reply:
I'm not super happy about this, but I don't have a better idea...